jbake-core/src/main/java/org/jbake/parser/AsciidoctorEngine.java

```diff
@@ -89,40 +89,57 @@ public void processHeader(final ParserContext context) {
         DocumentHeader header = asciidoctor.readDocumentHeader(context.getFile());
         Map<String, Object> contents = context.getContents();
         if (header.getDocumentTitle() != null) {
-        	contents.put("title", header.getDocumentTitle().getCombined());
+            contents.put("title", header.getDocumentTitle().getCombined());
         }
         Map<String, Object> attributes = header.getAttributes();
-        for (String key : attributes.keySet()) {
-            if (key.startsWith("jbake-")) {
-                Object val = attributes.get(key);
-                if (val!=null) {
-                    String pKey = key.substring(6);
-                    contents.put(pKey, val);
-                }
+        for (Map.Entry<String, Object> attribute : attributes.entrySet()) {
+            String key = attribute.getKey();
+            Object value = attribute.getValue();
+
+            if (hasJBakePrefix(key)) {
+                String pKey = key.substring(6);
+                contents.put(pKey, value);
             }
-            if (key.equals("revdate")) {
-                if (attributes.get(key) != null && attributes.get(key) instanceof String) {
-
-                    DateFormat df = new SimpleDateFormat(context.getConfig().getString(Keys.DATE_FORMAT));
-                    Date date = null;
-                    try {
-                        date = df.parse((String)attributes.get(key));
-                        contents.put("date", date);
-                    } catch (ParseException e) {
-                        e.printStackTrace();
-                    }
+            if (hasRevdate(key) && canCastToString(value)) {
+
+                String dateFormat = context.getConfig().getDateFormat();
+                DateFormat df = new SimpleDateFormat(dateFormat);
+                try {
+                    Date date = df.parse((String) value);
+                    contents.put("date", date);
+                } catch (ParseException e) {
+                    LOGGER.error("Unable to parse revdate. Expected {}", dateFormat, e);
                 }
             }
             if (key.equals("jbake-tags")) {
```

Comment:
Should we extract "jbake-tags" to a constant?

Reply:
Yes. There is a lot of stuff we should put into constants or find better substitutes.
I just wanted to focus on the configuration part with this PR.